src/tools/singlejar/diag.h

```diff
@@ -19,7 +19,7 @@
  * Various useful diagnostics functions from Linux err.h file, wrapped
  * for portability.
  */
-#if defined(__APPLE__) || defined(__linux)
+#if defined(__APPLE__) || defined(__linux__)
```

Comment:
On Power, when we compile a program using -std=c++11, then __linux gets undefined, which otherwise is defined. However, on x86, -std=c++11 does not make any difference. I verified on x86 too, and __linux__ is defined there as well.
On Power:
```
$ cpp -dM foo.c | grep linux
#define __linux 1
#define __linux__ 1
#define __gnu_linux__ 1
#define linux 1
$ cpp -dM -std=c++11 foo.c | grep linux
#define __linux__ 1
#define __gnu_linux__ 1
```

On x86:
```
$ cpp -dM foo.cc | grep linux
#define __linux 1
#define __linux__ 1
#define __gnu_linux__ 1
#define linux 1
~$ cpp -dM -std=c++11 foo.cc | grep linux
#define __linux 1
#define __linux__ 1
#define __gnu_linux__ 1

```